filter.go

```diff
@@ -128,6 +118,18 @@ func (api *API) Filters(ctx context.Context, rc *ResourceContainer, params Filte
 			params.ResultInfo = fResponse.ResultInfo.Next()
 		}
```

Comment:
@jacobbednarz this part of the code handles the multi-page condition. ^

The problem I intend to solve is a duplicate call of 
```
res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
```

When you call `Filters` like this:
```
var filterListParams = FilterListParams{
	PerPage: 25,
	Page:    1,
}

actual, _, err := client.Filters(context.Background(), ZoneIdentifier("d56084adb405e0b7e32c52321bf07be6"), filterListParams)
```